stackslib/src/net/relay.rs

```diff
@@ -845,6 +879,11 @@ impl Relayer {
             &obtained_method,
         );
 
+        #[cfg(any(test, feature = "testing"))]
```

Comment:
Minor nit -- instead of peppering the business logic with `#[cfg(...)]` statements, can you create two instances of `fault_injection::ignore_block()` -- one that compiles for testing, and one that's a no-op in production?

Reply:
Addressed in 76c91359c.